.ci/common.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# Attention, there is no "-x" to avoid problems on Travis
+set -e
+
+# result of command goes to $RUN_JOB
+# 0 = not run (skip)
+# 1 = run (no skip)
+function should_run_job {
+    SKIP_JOB_BY_FILES=$1
+    SKIP_FILES=$2
+    DEBUG=$3
+
+    echo "Initial value of SKIP_JOB_BY_FILES: $SKIP_JOB_BY_FILES"
+    echo "Files to that are allowed to be skipped: $SKIP_FILES"
+
+    if [[ $SKIP_JOB_BY_FILES != 'false' ]]; then
```

Comment:
blow mind .... as it is not real boolean, but string and ....

it is better to avoid boolean arguments , it is very non-intuitive, hard to read
Is it reasonable to refactor more and make it non boolean?  to make code something like `$MODE = 'skip_by_files'`

Reply:
I am not sure `mode` fits this. Currently this matches `Boolean` in java as we have "true", "false", and "recheck" which is more like `null`. "true" and "recheck" behave the same that if it is not "false" we run a recheck on if we should skip by files and then use that as the final condition of the function.

Reply:
I am not sure mode fits this. Currently this matches Boolean in java as we have "true", "false", and "recheck" which is more like null. "true" and "recheck" behave the same that if it is not "false" we run a recheck on if we should skip by files and then use that as the final condition of the function.

Reply:
Let's skip this for now, to not grow this PR,  we can improve later on